### ISSUE TITLE
bug: When `ParallelAgent` executes sub-agents in a session that persists across turns, subsequent user inputs are routed to only one sub-agent instead of the full parallel set

### ISSUE THREAD

**Issue by cloudy-ddyy** [zavora-ai/adk-rust#419]
## Description

When `ParallelAgent` executes sub-agents in a session that persists across turns, subsequent user inputs are routed to only one sub-agent instead of the full parallel set. The `find_agent_to_run` logic in the Runner scans past session events and routes to the last agent that responded, bypassing the `ParallelAgent` root.

## Steps to Reproduce

1. Create three `LlmAgent`s (`technical_analyst`, `business_analyst`, `ux_analyst`) and wrap them in a `ParallelAgent` named `multi_analyst`
2. Launch an interactive REPL via `Launcher` (which creates a single session and reuses it across all turns)
3. Enter any question as the first input — all three agents respond
4. Enter the same or a different question as the second input — only the last-responding agent (e.g., `ux_analyst`) responds; the other two are silent

## Expected Behavior

Every user input should trigger all sub-agents managed by the `ParallelAgent`, regardless of conversation turn or session history. The `ParallelAgent` root should always be selected as the target agent.

## Actual Behavior

After the first turn, `find_agent_to_run` scans session events in reverse, finds the last non-user event (authored by e.g., `ux_analyst`), and routes the next query directly to that single sub-agent. The `ParallelAgent` root is never consulted again, so its remaining sub-agents never execute.

The relevant framework code (`runner.rs:1112-1143`):

```rust
pub fn find_agent_to_run(root_agent, session) -> Arc<dyn Agent> {
    let events = session.events();
    for i in (0..events.len()).rev() {
        if let Some(event) = events.at(i) {
            if event.author == "user" { continue; }
            if let Some(agent) = Self::find_agent(root_agent, &event.author) {
                return agent;  // single sub-agent, not the ParallelAgent root
            }
        }
    }
    root_agent.clone()
}
```

This routing happens silently with no indication that agents are being skipped.

## Environment

- OS: Linux
- Rust version: stable (1.85+)
- ADK-Rust version: 1.0.0
- Affected crate(s): adk-runner, adk-agent

## Additional Context

The root cause is a design mismatch: the framework assumes a conversational turn-taking model where each new input should be handled by whatever agent last responded. `ParallelAgent` violates that assumption — it needs all sub-agents to run on every query.

Since `find_agent_to_run` lives in third-party framework code (`adk-runner-1.0.0/src/runner.rs`), it cannot be modified from the application layer.

Two application-level workarounds exist:

1. **New session per query**: Bypass `Launcher` and write a custom REPL loop that calls `SessionId::generate()` for each input. With an empty session, `find_agent_to_run` always falls back to the root `ParallelAgent`.
2. **Replace with `SequentialAgent`**: Run sub-agents sequentially instead of concurrently — no routing conflict, but changes execution semantics.

**Comment by jkmaina** [comment]
Resolved on main by commit 0c653af6. Runner resumption now checks supports_agent_transfer across the full agent path, so workflow roots such as ParallelAgent restart on each turn instead of routing to the last sub-agent. The regression test test_find_agent_to_run_resumes_workflow_root_not_subagent passes in the full workspace suite.
